keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEvent.kt

```diff
@@ -0,0 +1,28 @@
+package com.netflix.spinnaker.keel.lifecycle
+
+import java.time.Instant
+
+data class LifecycleEvent(
+  val scope: LifecycleEventScope,
+  val artifactRef: String,
+  val artifactVersion: String,
+  val type: LifecycleEventType,
+  val id: String? = null,
+  val status: LifecycleEventStatus,
+  val text: String? = null,
+  val link: String? = null,
+  val timestamp: Instant? = null
```

Comment:
It is added at the database saving level, so that consumers don't have to send it. but when I read the object I want to get it back. Do you think this is the best way to model that?

Reply:
Ohhhh... That makes sense.